trailblazer/store/api.py

```diff
@@ -111,29 +108,27 @@ def set_analysis_completed(self, analysis_id: int) -> None:
 
     def set_analysis_uploaded(self, case_id: str, uploaded_at: dt.datetime) -> None:
         """Setting analysis uploaded at."""
-        analysis_obj: Analysis = self.get_latest_analysis(case_id=case_id)
-        analysis_obj.uploaded_at: dt.datetime = uploaded_at
+        analysis: Optional[Analysis] = self.get_latest_analysis_for_case(case_name=case_id)
+        analysis.uploaded_at = uploaded_at
         self.commit()
 
     def set_analysis_status(self, case_id: str, status: str):
         """Setting analysis status."""
-        status = status.lower()
+        status: str = status.lower()
         if status not in set(TrailblazerStatus.statuses()):
             raise ValueError(f"Invalid status. Allowed values are: {TrailblazerStatus.statuses()}")
-        analysis_obj: Analysis = self.get_latest_analysis(case_id=case_id)
-        analysis_obj.status: str = status
+        analysis: Optional[Analysis] = self.get_latest_analysis_for_case(case_name=case_id)
+        analysis.status = status
         self.commit()
-        LOG.info(f"{analysis_obj.family} - Status set to {status.upper()}")
+        LOG.info(f"{analysis.family} - Status set to {status.upper()}")
 
     def add_comment(self, case_id: str, comment: str):
```

Comment:
Probably out of scope for this PR but would we want to match so all "case_id" switch to "case_name"?

Reply:
Another PR